referenceframe/frame.go

```diff
@@ -161,28 +195,23 @@ func NewStaticFrameFromFrame(frame Frame, pose spatial.Pose) (Frame, error) {
 	if pose == nil {
 		return nil, errors.New("pose is not allowed to be nil")
 	}
-	if tf, ok := frame.(*translationalFrame); ok {
-		return &staticFrame{tf.Name(), pose, tf.geometryCreator}, nil
-	}
-	if tf, ok := frame.(*staticFrame); ok {
-		return &staticFrame{tf.Name(), pose, tf.geometryCreator}, nil
+	switch f := frame.(type) {
+	case *staticFrame:
```

Comment:
Nit: Combine the cases and cut down on the number of return statements?

Reply:
I tried this and due to the way Go works, it wasn't really obvious how this would be accomplished.  For a combined case, the variable `f` does not know which case it is so it can't say for sure if there is a `geometryCreator` in the given class (since its not a public member).  Open to suggestions if you see a better way to write this though!